assets/src/components/lateView.tsx

```diff
@@ -261,4 +528,112 @@ const MissingLogonRow = ({
   )
 }
 
+const HideCheckbox = ({
+  vehicleOrGhost,
+}: {
+  vehicleOrGhost: VehicleOrGhost
+}): ReactElement<HTMLElement> | null => {
+  const { selectedIds, toggleCheckedState } = useContext(LateViewContext)
+  const isChecked = vehicleOrGhost.runId
+    ? idIn(vehicleOrGhost.runId, selectedIds)
+    : false
+
+  return vehicleOrGhost.runId ? (
+    <input
+      type="checkbox"
+      checked={isChecked}
+      onClick={() => toggleCheckedState(vehicleOrGhost.runId!)}
+    />
+  ) : null
+}
+
+const HidePopup = ({
+  nRowsSelected,
+  hideSelectedRows,
+}: {
+  nRowsSelected: number
+  hideSelectedRows: () => void
+}) => (
+  <div className="m-late-view__popup">
+    {nRowsSelected} selected
+    <button onClick={hideSelectedRows}>Hide</button>
+  </div>
+)
+
+const UnhidePopup = ({
+  nRecentlyHidden,
+  unhideRecentlyHidden,
+}: {
+  nRecentlyHidden: number
+  unhideRecentlyHidden: () => void
+}): ReactElement<HTMLElement> => (
+  <div className="m-late-view__popup">
+    {nRecentlyHidden} hidden
+    <button onClick={unhideRecentlyHidden}>Undo</button>
+  </div>
+)
+
+const UnhideToggle = ({
+  viewHidden,
+  toggleViewHidden,
+}: {
+  viewHidden: boolean
+  toggleViewHidden: () => void
+}): ReactElement<HTMLElement> => (
+  <button
+    onClick={() => {
+      if (window && window.FS) {
+        window.FS.event("User clicked eye toggle")
+      }
+      toggleViewHidden()
+    }}
+  >
+    {viewHidden ? "Hide" : "Show"}
+  </button>
+)
+
+const MasterCheckbox = ({
+  attachedIds,
+}: {
+  attachedIds: RunId[]
+}): ReactElement<HTMLElement> => {
+  const { selectedIds, setSelectedIds } = useContext(LateViewContext)
+
+  const selectedAttachedIds = selectedIds.filter((id) =>
+    attachedIds.includes(id)
+  )
+  const selectionState =
```

Comment:
**suggestion (non-blocking):** this is probably typed as `String` and, if we wanted to be super pedantic, could be explicitly typed as `"none" | "all" | "some"`.